### PR TITLE
Update xcdat_open()

### PR DESCRIPTION
**Pull request by acordonez**
1. Pass "chunks" parameter to xcdat_open so that drivers can set chunk size if needed. Default is {} (see [xarray.open_dataset() documentation](https://docs.xarray.dev/en/stable/generated/xarray.open_dataset.html) for more details)
2. Catch cases where dataset cannot be opened because of non-cf compliant attributes. Currently only handles the case where the calendar name contains "-" instead of "_". Can be easily expanded to handle more cases.

xcdat_open is currently used by the mean climate metrics, the modes of variability metrics, mjo metrics, and monsoon (Sperber).

### CODE REVIEW

**Comment by acordonez** [comment]
@lee1043 This is low priority to review. For testing I've run Demos 1b, 2b, 4, and 5 as it looks like those drivers use the xcdat_open() function.

**Review by lee1043 — APPROVED** [review]
Looks good to me. Also checked with demo 4 notebook which ran without any issue. Thanks for the PR, @acordonez!

File: pcmdi_metrics/io/xcdat_openxml.py [hunk 1]
```diff
@@ -7,9 +7,11 @@
 import xcdat as xc
 import xmltodict
 
+from pcmdi_metrics.io.xcdat_dataset_io import get_calendar
+
 
 def xcdat_open(
-    infile: Union[str, list], data_var: str = None, decode_times: bool = True
+    infile: Union[str, list], data_var: str = None, decode_times: bool = True, chunks={}
```
[the remark this answers is not in the document]
Comment (acordonez) [reply pcmdi_metrics/io/xcdat_openxml.py:14]: @lee1043 The extremes and drcdm metrics need to be able to specify the chunks to ensure that the time axis is continuous across a single chunk.

[the remark this answers is not in the document]
Comment (acordonez) [reply pcmdi_metrics/io/xcdat_openxml.py:14]: @lee1043 Would chunks=None mean that no chunking is used by default? That might be fine?

[the remark this answers is not in the document]
Comment (lee1043) [reply pcmdi_metrics/io/xcdat_openxml.py:14]: > @lee1043 The extremes and drcdm metrics need to be able to specify the chunks to ensure that the time axis is continuous across a single chunk.

That sounds like a good reason to keep the PR as it is. Thanks for the comment!

[the remark this answers is not in the document]
Comment (lee1043) [reply pcmdi_metrics/io/xcdat_openxml.py:14]: @acordonez I just found that `chunks={}` as default causes error in modes of variability code by opening dataset with dask chunks. I can make modes of variability as a special case by having `chunks=None` when using `xcdat_open` but haven't test with other metrics. If dask chunks are needed for only a few metrics, how about setting the default as `chunks=None` while in those special cases use `xcdat_open` with `chunks={}`?

[the remark this answers is not in the document]
Comment (acordonez) [reply pcmdi_metrics/io/xcdat_openxml.py:14]: @lee1043 That suggestions sounds good to me.

**Pull request merged**